csi/src/nodeplugin_grpc.rs

```diff
@@ -0,0 +1,100 @@
+//! The mayastor node plugin gRPC service
+//! This provides access to functionality that needs to be executed on the same
+//! node as a Mayastor CSI node plugin, but it is not possible to do so within
+//! the CSI framework. This service must be deployed on all nodes the the
+//! Mayastor CSI node plugin is deployed.
+use crate::freezefs;
+use freezefs::{freeze_volume, unfreeze_volume, FreezeFsError};
+use mayastor_node_plugin::*;
+use tonic::{transport::Server, Code, Request, Response, Status};
+
+pub mod mayastor_node_plugin {
+    tonic::include_proto!("mayastornodeplugin");
+}
+
+#[derive(Debug, Default)]
+pub struct MayastorNodePluginSvc {}
+
+impl From<FreezeFsError> for Status {
+    fn from(err: FreezeFsError) -> Self {
+        let msg = format!("{:?}", err);
+        match err {
+            FreezeFsError::VolumeNotFound {
+                volid: _,
+            } => Status::new(Code::NotFound, msg),
+            FreezeFsError::FsfreezeFailed {
+                volid: _,
+                error: _,
+            } => Status::new(Code::Internal, msg),
+            FreezeFsError::InvalidVolumeId {
+                volid: _,
+                source: _,
+            } => Status::new(Code::InvalidArgument, msg),
+            FreezeFsError::InternalFailure {
+                volid: _,
+                source: _,
+            } => Status::new(Code::Internal, msg),
+            FreezeFsError::IOError {
+                volid: _,
+                source: _,
+            } => Status::new(Code::Unknown, msg),
+        }
+    }
+}
+
+#[tonic::async_trait]
+impl mayastor_node_plugin_server::MayastorNodePlugin for MayastorNodePluginSvc {
```

Comment:
theseAreRatherLongNamesIsThatNeeded?

Reply:
The long name is a consequence of this line in the .proto file 
```
service MayastorNodePlugin { 
```
I elected for a descriptive name for the service as opposed to a shorter more cryptic one.